### DISCUSSION TITLE
ZK compression

### DISCUSSION THREAD

**Discussion by kimurayu45z** [cosmos/community#86] [Cosmos SDK]
Solana introduced ZK compression.
https://x.com/LightProtocol/status/1804241026334548366

Cosmos SDK possess logics natively (non VM) so there would be a difficulty to generate the ZKP of the state transition, but I think the community should discuss and investigate a way to follow such latest movement.

This also may lead to the capability of building zk sovereign rollup with Rollkit.

**Comment by tac0turtle** [comment]
thank you for opening these discussions, really loving them. Keep them coming!!

We have been wokring with risc0 on provng execution, we recently ran into an issue with it and are evaluating how to best move forward with it.
